File: Contribution/BinarySearch.cpp
int binarySearch(int arr[], int left, int right, int n)
{
    while (left <= right)
    {
        int mid = left + (right - left) / 2;
 
        if (arr[mid] == n)
            return mid;
 
        if (arr[mid] < n)
            left = mid + 1;
 
        else
            right = mid - 1;
    }
    return -1;
}
 

File: Contribution/BinarySearch.js
function binarySearch(arr, l, r, x) {
    if (r >= l) {
        let mid = l + Math.floor((r - l) / 2);

        if (arr[mid] == x)
            return mid;

        if (arr[mid] > x)
            return binarySearch(arr, l, mid - 1, x);

        return binarySearch(arr, mid + 1, r, x);
    }
 
    return -1;
}

// Source: GeekForGeeks
